src/app/shared/services/base.service.spec.ts

```diff
@@ -0,0 +1,94 @@
+import { TestBed } from '@angular/core/testing';
+import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
+
+import { environment } from '../../../environments/environment';
+import { BaseService } from './base.service';
+import { HttpErrorResponse } from '@angular/common/http';
+
+describe('BaseService', () => {
+  let service: BaseService;
+  let httpMock: HttpTestingController;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        HttpClientTestingModule
+      ]
+    });
+    service = TestBed.inject(BaseService);
+    httpMock = TestBed.inject(HttpTestingController);
+  });
+
+  it('creates the service', () => {
+    expect(service).toBeTruthy();
+  });
+
+  describe('#post', () => {
+    describe('Given a valid data', () => {
```

Comment:
`data` is always too generic and not precise. So I usually recommend o stay clear from it unless there are no better options. Here I find it confusing as `data` could refer to both the JSON API attribute and request body. How about to clarify with:

```suggestion
    describe('Given a valid request payload', () => {
```